commands.js

```diff
@@ -1448,6 +1448,10 @@ exports.commands = {
 		if (room.modchat && room.modchat.length <= 1 && Config.groupsranking.indexOf(room.modchat) > 1 && !user.can('modchatall', null, room)) {
 			return this.errorReply("/modchat - Access denied for removing a setting higher than " + Config.groupsranking[1] + ".");
 		}
+		if (room.battle) {
```

Comment:
Because it's not proper duck typing.
It's "if this object has this function, call it", not "if this object is of this type, call this function"